types/serverless/index.d.ts

```diff
@@ -0,0 +1,76 @@
+// Type definitions for serverless 1.18
+// Project: https://github.com/serverless/serverless#readme
+// Definitions by: Hassan Khan <https://github.com/hassankhan>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import Service = require("./classes/service");
+
+declare namespace Serverless {
+    interface Options {
+        stage: string | null;
+        region: string | null;
+        noDeploy?: boolean;
+    }
+
+    namespace Provider {
+        class Aws {
+            constructor(serverless: Serverless, options: Options)
+
+            getProviderName(): string;
+            getRegion(): string;
+            getServerlessDeploymentBucketName(): string;
+            getStage(): string;
+        }
+    }
+
+    namespace Service {
```

Comment:
You can move this interface into the service class. Avoid having serverless namespace all together

Reply:
Would a user of the types package still be able to use ambient declarations to override the `Custom` interface? 

The use-case was mainly for framework plugin authors, since most plugins take their configuration from the `custom` property in the `Service` class.